src/Gui/propertyeditor/PropertyItem.cpp

```diff
@@ -2844,16 +2848,35 @@ void PropertyEnumItem::setValue(const QVariant& value)
     setPropertyValue(data);
 }
 
-QWidget* PropertyEnumItem::createEditor(QWidget* parent, const QObject* receiver, const char* method) const
+class EnumItems;
+
+struct EnumItem {
+    QString text;
+    QString fullText;
+    std::shared_ptr<EnumItems> children;
+    EnumItem(const QString &t = QString(), const QString &f = QString())
+        :text(t), fullText(f)
+    {}
+    void populate(QMenu *menu);
+};
+
+class EnumItems : public std::vector<EnumItem>

```

Comment:
Is there a reason to leave this a full-blown class, and not just a `typedef`?

Reply:
Because it needs to be forward declared for use by `EnumItem::children`.